### DISCUSSION TITLE
Feature Idea: Using BL Touch to guide better manual leveling

### DISCUSSION THREAD

**Discussion by samsp-msft** [Jyers/Marlin#329] [Ideas]
While I understand that the printer can account for an inaccurate bed level via the mesh, it seems to me that setting the spring height to be as close to even as possible would probably not be a bad thing. After doing a bed level you can get a report of what the offsets are for each measuring point. I wonder if that could be translated into what adjustments need to be made to the spring tensioners to even it out?

What I am imagining is a mode, where you run a bed level operation, and then the UI reports how much and which direction each corner needs to be adjusted. For example:
Bottom Left: None
Bottom Right: 1.5 turn :leftwards_arrow_with_hook:
Top Left: 0.25 turn  :arrow_right_hook:
Top Right: 2.25 turn :leftwards_arrow_with_hook:

Which would bring the bed closer to level. It may need to be repeated a couple of times to narrow it down, but would bring the bed to close to perpendicular so the mesh would mainly be about accounting for unevenness in the build surface. This seems like it would be more accurate than with paper, and much easier to use.

**Comment by xantrk** [comment]
I believe there is already a marlin feature for this. 
G35 - Tramming Assistant
Would be really nice to be able to do it from LCD

**Comment by Fury1SOG** [comment]
This would be neat to have in the firmware.
In the meantime, if you use Octoprint, a plugin called AutoBim is available that does this: 
https://github.com/j-be/AutoBim

**Comment by samsp-msft** [comment]
I have built my own build of Marlin enabling the G35 Tramming option. I think it has a lot of potential for helping to set an initial bed position that is close to perfect.

I had 2 issues:
- The default probing is 20, 180mm which is not well aligned with the bed screws which are at ~33 and 203mm.  My probe offset is approx -43, -8mm, even setting a probe margin to 0, the build will fail due to a coordinate check as it won't go outside the bed bounds for the nozzle. The X gantry has enough room for the probe to reach the screw positions without hitting the stop - but I am not sure what can be adjusted in marlin to enable it to measure at those points.
- I checked the option to Show Tramming in the UI, but it didn't appear anywhere I could find. I used it by sending the G35 command via proterface and it measured the 4 points and gave instructions on which screws to adjust and how many turns/minutes. If that info could be put on the display, it would be the finishing touch.

**Comment by samsp-msft** [comment]
I have fixed the first issue by changing X_MAX_POS to 253 - this value may depend on what modifications you have done to your printer - this is where a bolt for the x gantry would hit the bracket on my printer - I have a dual z stepper configuration so I can't remember if that bracket was changed.
G35 will now probe over the bed screw positions, and report back the turns that are needed, and it seems to be accurate.

I am unsure of why problem 2 happens. https://github.com/MarlinFirmware/Marlin/pull/20000 should have added the UI, but I wonder if there is a difference between which UI that was added for. It seems to be affecting a different config from creality_dwin.cpp. @Jyers can you point me in the right direction?

**Comment by rorywquin** [comment]
If you use Octoprint the "Bed Visualiser" plugin has this feature - i.e. tells you how much to turn the adjustment wheel at each corner
